lib/jellyfish/room.ex

```diff
@@ -92,7 +94,19 @@ defmodule Jellyfish.Room do
   @spec add_component(id(), Component.component(), map()) ::
           {:ok, Component.t()}
           | :error
-          | {:error, :incompatible_codec | :reached_components_limit_hls}
+          | {:error,
+             {:component_disabled_globally, String.t()}
+             | :incompatible_codec
+             | {:reached_components_limit, String.t()}
+             | :file_does_not_exist
+             | :bad_parameter_framerate_for_audio
+             | :invalid_framerate
+             | :invalid_file_path
+             | :unsupported_file_type
+             | {:missing_parameter, term()}
+             | :missing_s3_credentials
+             | :overriding_credentials
+             | :overriding_path_prefix}
```

Comment:
Lol a lot of new errors. I assume the previous spec wasn't updated in a long time 😄